tests/unit-core/test-number-converter.c

```diff
@@ -0,0 +1,128 @@
+} /* test_to_uint32 */
+
+// basic toInt32 tester method
+static void
+test_to_int32 (double input, int32_t test_number)
+{
+  jerry_value_t number_val = jerry_create_number (input);
+  int32_t int_number;
+  number_val = jerry_value_as_int32 (number_val, &int_number);
+  TEST_ASSERT (!jerry_value_is_error (number_val) && int_number == test_number);
+} /* test_to_int32 */
+
+// basic toInteger tester method
+static void
+test_to_interger (double input, double test_number)
+{
+  jerry_value_t number_val = jerry_create_number (input);
+  double double_number;
+  number_val = jerry_value_as_integer (number_val, &double_number);
+  TEST_ASSERT (!jerry_value_is_error (number_val) && double_number == test_number);
+} /* test_to_interger */
+
+int
+main (void)
+{
+  TEST_INIT ();
+
+  jerry_init (JERRY_INIT_EMPTY);
+
+  // few toUint32 test-cases
+  test_to_uint32 (1.0, 1);
+  test_to_uint32 (0.0, 0);
+  test_to_uint32 (NAN, 0);
+  test_to_uint32 (-NAN, 0);
+  test_to_uint32 (INFINITY, 0);
+  test_to_uint32 (-INFINITY, 0);
+  test_to_uint32 (0.1, 0);
+  test_to_uint32 (-0.1, 0);
+  test_to_uint32 (1.1, 1);
+  test_to_uint32 (-1.1, 4294967295);
+  test_to_uint32 (4294967295, 4294967295);
+  test_to_uint32 (-4294967295, 1);
+  test_to_uint32 (4294967296, 0);
+  test_to_uint32 (-4294967296, 0);
+  test_to_uint32 (4294967297, 1);
+  test_to_uint32 (-4294967297, 4294967295);
+
+  // few toint32 test-cases
+  test_to_int32 (1.0, 1);
+  test_to_int32 (0.0, 0);
+  test_to_int32 (NAN, 0);
+  test_to_int32 (-NAN, 0);
+  test_to_int32 (INFINITY, 0);
+  test_to_int32 (-INFINITY, 0);
+  test_to_int32 (0.1, 0);
+  test_to_int32 (-0.1, 0);
+  test_to_int32 (1.1, 1);
+  test_to_int32 (-1.1, -1);
+  test_to_int32 (4294967295, -1);
+  test_to_int32 (-4294967295, 1);
+  test_to_int32 (4294967296, 0);
+  test_to_int32 (-4294967296, 0);
+  test_to_int32 (4294967297, 1);
+  test_to_int32 (-4294967297, -1);
+  test_to_int32 (2147483648, -2147483648);
+  test_to_int32 (-2147483648, -2147483648);
+  test_to_int32 (2147483647, 2147483647);
+  test_to_int32 (-2147483647, -2147483647);
+  test_to_int32 (-2147483649, 2147483647);
+  test_to_int32 (2147483649, -2147483647);
+
+  // few toInteger test-cases
+  test_to_interger (1.0, 1.0);
+  test_to_interger (0.0, 0.0);
+  test_to_interger (NAN, 0);
+  test_to_interger (-NAN, 0);
+  test_to_interger (INFINITY, INFINITY);
+  test_to_interger (-INFINITY, -INFINITY);
+  test_to_interger (0.1, 0);
+  test_to_interger (-0.1, -0);
+  test_to_interger (1.1, 1);
+  test_to_interger (-1.1, -1);
+  test_to_interger (4294967295, 4294967295);
+  test_to_interger (-4294967295, -4294967295);
+  test_to_interger (4294967295, 4294967295);
+  test_to_interger (-4294967296, -4294967296);
+  test_to_interger (4294967297, 4294967297);
+  test_to_interger (-4294967297, -4294967297);
+
+  // few test-cases which return with error
+  jerry_value_t error_val =  jerry_create_error (JERRY_ERROR_TYPE, (const jerry_char_t *) "error");
```

Comment:
Please add tests for an object like this: `{ valueOf() { throw new TypeError("foo")}}`. You can evaluate this string to construct this object more easily.